src/transformStateWithClones.js

```diff
@@ -6,8 +6,32 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let resultObject = { ...state };
+  const resultArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
```

Comment:
```suggestion
  for (const action of actions) {
```

You are not using indexes in your code so we can use for of loop